epaint/src/text/font.rs

```diff
@@ -282,9 +343,34 @@ impl Font {
         let mut cursor_x_in_points = 0.0f32;
         let mut last_glyph_id = None;
 
+        let mut tab_size = super::MAX_TAB_SIZE;
+
         for c in text.chars() {
             if !self.fonts.is_empty() {
-                let (font_index, glyph_info) = self.glyph_info(c);
+                // For tabs it is nice to let them occupy only what's left of the current
+                // tab block. Better to show you as I don't know how to describe it in words
+                //
+                // We consider the following sequence of characters, separated with spaces for clarity:
+                // \t 1 \t 12 \t 123 \t 1234 \t
+                //
+                // It will be rendered like this, using ---> to denote the space occupied by a tab:
+                //
+                // 1234 | 1234 | 1234 | 1234 | 1234
+                // -----+------+------+------+-----
+                // ---> | 1--> | 12-> | 123> | --->
+                let (font_index, glyph_info) = if c == '\t' {
+                    let result = self.tab_glyph_info(tab_size);
+                    tab_size = super::MAX_TAB_SIZE;
+                    result
+                } else {
+                    tab_size -= 1;
+                    self.glyph_info(c)
+                };
```

Comment:
A lot simpler implementation of this would be to just modify `glyph_info.advance_width` directly here. No need for `fn tab_glyph_info` and a separate glyph cache for different tab widths